example/screens/ConnectWithRoomManagerScreen.tsx

```diff
@@ -0,0 +1,152 @@
+import React, {useState} from 'react';
+import {
+  Dimensions,
+  Image,
+  SafeAreaView,
+  StyleSheet,
+  Text,
+  View,
+} from 'react-native';
+import {connectScreenLabels} from '../types/ComponentLabels';
+import {useJellyfishClient} from '@jellyfish-dev/react-native-client-sdk';
+import {Button, TextInput, DismissKeyboard} from '../components';
+import {NativeStackScreenProps} from '@react-navigation/native-stack';
+import {AppRootStackParamList} from '../navigators/AppNavigator';
+import type {BottomTabScreenProps} from '@react-navigation/bottom-tabs';
+import {TabParamList} from '../navigators/AppNavigator';
+import {CompositeScreenProps} from '@react-navigation/native';
+import {usePermissionCheck} from '../hooks/usePermissionCheck';
+
+type Props = CompositeScreenProps<
+  BottomTabScreenProps<TabParamList, 'ConnectWithRoomManager'>,
+  NativeStackScreenProps<AppRootStackParamList>
+>;
+
+const {URL_INPUT, TOKEN_INPUT, CONNECT_BUTTON} = connectScreenLabels;
+
+async function getJellyFishServer(
+  roomManagerUrl: string,
+  roomName: string,
+  userName: string,
+) {
+  const url = roomManagerUrl.endsWith('/')
+    ? roomManagerUrl
+    : roomManagerUrl + '/';
+  const result = await fetch(
+    `${url}rooms/${roomName.trim()}/users/${userName.trim()}`,
+  );
+
+  const tokenData = (await result.json()) as {
+    url: string;
+    token: string;
+  };
+
+  return {
+    jellyfishUrl: tokenData.url,
+    token: tokenData.token,
+  };
+}
+
+const ConnectScreen = ({navigation}: Props) => {
+  const {connect} = useJellyfishClient();
+  const [connectionError, setConnectionError] = useState<string | undefined>(
+    undefined,
```

Comment:
nit: for values that are set explicitly, I'd rather expect `null` than `undefined`
```suggestion
  const [connectionError, setConnectionError] = useState<string | null>(
    null,
```

Reply:
We had some interesting discussion about `null` vs `undefined` and decided to go for `undefined` in such cases. But seems like `null` is more common in this codebase indeed, so I will update.